### ISSUE TITLE
Search function resets language preference

### ISSUE THREAD

**Issue by michael-lavaveshkul** [php/web-php#435]
From manual page: https://php.net/function.array-walk

---

First time submitting a bug to this project. I'm not sure which project this should go under, but I just clicked the "Report a bug" link from the docs.

1. Go to a page in the documentation where you can change the language.
2. Set browser language to something other than your browser's default language.
   1. For example, my browser is set to Japanese, but I chose to use the documentation in English.
3. Go to the search bar and type in a valid top level function name and hit enter. E.g. "count". (Do not click something from the dropdown.)
**Expected**: The resulting page is in the language I set (English).
**Actual**: The resulting page is in Japanese.
After that, clicking links will also stay in Japanese until I set my language again.

Similar scenario but with a help page as an intermediary.
1. Go to a page in the documentation where you can change the language.
2. Set browser language to something other than your browser's default language.
   1. For example, my browser is set to Japanese, but I chose to use the documentation in English.
3. Go to the search bar and type in an invalid top level function name and hit enter. E.g. "impld". (Do not click something from the dropdown.)
4. In the resulting help page, it will say "**impld** doesn't exist. Closest matches:" in English (regardless of your chosen language preference or browser's default language I think).
5. Click on one of the listed matches.
**Expected**: The resulting page is in the language I set (English).
**Actual**: The resulting page is in Japanese.
After that, clicking links will also stay in Japanese until I set my language again.

Clicking results from the dropdown doesn't have this issue.

**Comment by cmb69** [comment]
Not sure what to do here. If we prefer GH issues for Website related bugs, then https://bugs.php.net/bug.php?id=68880 should be closed as duplicate.

**Comment by salathe** [comment]
> Not sure what to do here. If we prefer GH issues for Website related bugs …

Has switching away from bugsweb to GitHub issues been discussed, and received general mumblings of approval, on the webmaster list?  I was surprised to see today that GitHub issues has been enabled for this repo, but maybe I missed the discussion.

**Comment by cmb69** [comment]
> […], but maybe I missed the discussion.

No, you have not. I've written to the ML a few hours ago, but can't link, since news.php.net is down, and marc.info apparently doesn't have the webmaster's list.

**Comment by saundefined** [comment]
@michael-lavaveshkul

Hi, after updating the search and navbar, I guess the issue is no longer relevant, and we can close it

**Comment by michael-lavaveshkul** [comment]
Hi @saundefined,

I think I found something related to my second scenario from my original post. The language selected isn't respected in the search results.

1. I set my browser's default language to Japanese first, English second.
2. I navigated to [the documentation page](https://www.php.net/docs.php)
3. Then opened [the Spanish docs](https://www.php.net/manual/es/)
4. I Went to the search bar and searched for text `curl_close` and hit enter, directing me to [https://www.php.net/search.php?lang=es&q=curl_close](https://www.php.net/search.php?lang=es&q=curl_close).

**Expected**: The google results would be in Spanish. Especially since `lang=es` is part of the query string.

**Actual**:  It seems to give me results in all the languages in All Results, defaulting to Japanese and English first (based on my browser?).

![Image](https://github.com/user-attachments/assets/edd79aa4-96dd-4b10-94bf-9274dfb3a7bc)

I'm fine to close this one if it's out of scope.

**Comment by sy-records** [comment]
@michael-lavaveshkul Hi, Google results are beyond our control. By default, it displays results in all languages. When you search for `curl_close`, you will see multiple identical results, but in different languages. If you need to search for a specific language, you can add `"lang"`, for example: `curl_close "es"`
